src/table/resizer/index.tsx

```diff
@@ -97,83 +108,138 @@ export function Resizer({
+  const resizerAriaProps =
+    resizerRole === 'button'
+      ? {
+          'aria-labelledby': joinStrings(ariaLabelledby, resizerWidthId),
+          'aria-pressed': false,
         }
-        event.preventDefault();
-        const headerCell = findUpUntil(event.currentTarget, element => element.tagName.toLowerCase() === 'th')!;
-        setIsDragging(true);
-        setHeaderCell(headerCell);
-      }}
-      onFocus={event => {
-        const headerCell = findUpUntil(event.currentTarget, element => element.tagName.toLowerCase() === 'th')!;
-        setHeaderCellWidth(headerCell.getBoundingClientRect().width);
-        setResizerHasFocus(true);
-        setHeaderCell(headerCell);
-        onFocus?.();
-      }}
-      onBlur={() => {
-        setResizerHasFocus(false);
-        onBlur?.();
-      }}
-      role="separator"
-      aria-orientation="vertical"
-      aria-labelledby={ariaLabelledby}
-      aria-valuenow={headerCellWidth}
-      // aria-valuetext is needed because the VO announces "collapsed" when only aria-valuenow set without aria-valuemax
-      aria-valuetext={headerCellWidth.toString()}
-      aria-valuemin={minWidth}
-      tabIndex={tabIndex}
-      data-focus-id={focusId}
-    />
+      : {
+          'aria-labelledby': ariaLabelledby,
+          'aria-orientation': 'vertical' as const,
+          'aria-valuenow': headerCellWidth,
+          // aria-valuetext is needed because the VO announces "collapsed" when only aria-valuenow set without aria-valuemax
+          'aria-valuetext': headerCellWidthString,
+          'aria-valuemin': minWidth,
+        };
+
+  return (
+    <>
+      <span
+        className={clsx(
+          styles.resizer,
+          isDragging && styles['resizer-active'],
+          (resizerHasFocus || showFocusRing) && styles['has-focus']
+        )}
+        onMouseDown={event => {
+          if (event.button !== 0) {
+            return;
+          }
+          event.preventDefault();
+          const headerCell = findUpUntil(event.currentTarget, element => element.tagName.toLowerCase() === 'th')!;
+          setIsDragging(true);
+          setHeaderCell(headerCell);
+        }}
+        onFocus={event => {
+          const headerCell = findUpUntil(event.currentTarget, element => element.tagName.toLowerCase() === 'th')!;
+          setHeaderCellWidth(headerCell.getBoundingClientRect().width);
+          setResizerHasFocus(true);
+          setHeaderCell(headerCell);
+          onFocus?.();
+        }}
+        onBlur={() => {
+          setResizerHasFocus(false);
+          onBlur?.();
+          if (isKeyboardDragging) {
+            setIsKeyboardDragging(false);
+            handlers?.resetColumnWidth();
+          }
+        }}
+        role={resizerRole}
+        {...resizerAriaProps}
+        tabIndex={tabIndex}
+        data-focus-id={focusId}
+      />
+      <ScreenreaderOnly id={resizerWidthId}>{headerCellWidthString}</ScreenreaderOnly>
```

Comment:
Is the 'aria-valuetext' announcing it already? Is it because of changing to button role?

Reply:
Yes, the `aria-valuetext` is only added when the role is "separator".

Reply:
I think it makes more sense to me to announce it after "Left"/"Right" key down, to give user more instant feedback when resizing. So we do not need to announce it after the resize is over.

Reply:
Looks like the aria-valuetext is not announced instantly when resizing using keyboard.

Reply:
I observe that more often than not with Chrome+VO but it is consistent with the current behaviour.

Reply:
You are right, looks good on Safari + VO.